plugins/partiql-local/src/main/kotlin/org/partiql/plugins/local/LocalCatalog.kt

```diff
@@ -0,0 +1,70 @@
+package org.partiql.plugins.local
+
+import com.amazon.ionelement.api.loadSingleElement
+import org.partiql.spi.BindingPath
+import org.partiql.types.StaticType
+import java.io.File
+import java.nio.file.Path
+
+private sealed class FsTree(val name: String) {
+
+    // "Directory" node
+    class D(name: String, val children: List<FsTree>) : FsTree(name)
+
+    // Type node
+    class T(name: String, val type: StaticType) : FsTree(name)
+}
+
+/**
+ * Build a memoized catalog tree from local schema definitions.
+ */
+public class LocalCatalog private constructor(private val root: FsTree.D) {
+
+    /**
+     * Search the tree for the type.
+     */
+    public fun lookup(path: BindingPath): LocalObject? {
+        val match = mutableListOf<String>()
+        var curr: FsTree? = root
+        for (step in path.steps) {
+            if (curr == null) return null
+            when (curr) {
+                is FsTree.T -> break
+                is FsTree.D -> {
+                    curr = curr.children.firstOrNull { step.isEquivalentTo(it.name) }
+                    if (curr != null) match.add(curr.name)
+                }
+            }
+        }
+        // All steps matched and we're at a leaf
+        if (curr is FsTree.T) {
```

Comment:
There is a potential bug here, but it's actually undefined behavior. Is it possible for a schema and object to have the same name? ie should this file structure be possible

```ion
catalog/
    ├── example
    │   └── other.ion
    └── example.ion
```